test/CloudNative.CloudEvents.IntegrationTests/AspNetCore/CloudEventControllerTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Cloud Native Foundation. 
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+namespace CloudNative.CloudEvents.IntegrationTests.AspNetCore
+{
+    using System;
+    using System.Net;
+    using System.Threading.Tasks;
+    using CloudNative.CloudEvents.AspNetCoreSample;
+    using Microsoft.AspNetCore.Mvc.Testing;
+    using Xunit;
+
+    public class CloudEventControllerTests : IClassFixture<WebApplicationFactory<Startup>>
+    {
+        private readonly WebApplicationFactory<Startup> _factory;
+
+        public CloudEventControllerTests(WebApplicationFactory<Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        [Fact]
+        public async Task Controller_WithValidCloudEvent_DeserializesUsingPipeline()
+        {
+            // Arrange
+            var cloudEvent = new CloudEvent("test-type", new Uri("urn:integration-tests"))
+            {
+                Id = Guid.NewGuid().ToString(),
+            };
+            var content = new CloudEventContent(cloudEvent, ContentMode.Structured, new JsonEventFormatter());
+
+            // Act
+            var result = await _factory.CreateClient().PostAsync("/api/events/receive", content);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            Assert.Contains(cloudEvent.Id, await result.Content.ReadAsStringAsync());
```

Comment:
It would probably be valuable to verify some other aspect of the data you're sending across other than one of the simpler properties.

What could be happening in your case:
1. Your formatter doesn't run because the content type is `application/cloudevents+json` and your formatter is looking for `application/cloudevents`
2. The vanilla MVC JSON formatter runs and successfully deserializes 
3. You're verifying the `id` property that doesn't require any specialized logic in order to deserialize

It's possible based on my guess that you're formatter isn't running in this case but the test still passes.